Source/NSURL.m

```diff
@@ -2256,12 +2265,10 @@ - (instancetype) init
 
 - (instancetype) initWithString: (NSString *)URLString
 {
-  self = [self init];
-  if (self != nil)
-    {
-      [self setString: URLString];
-    }
-  return self;
+  //OSX behavior is to return nil for a string which cannot be used to initialize valid NSURL object
+  NSURL* url = [NSURL URLWithString:URLString];
+  if(url) return [self initWithURL:url resolvingAgainstBaseURL:NO];
```

Comment:
Could you please use curly brackets here?

Reply:
Ok